File: backend/src/main/java/com/hst/sitescrapper/controller/ScrapController.java
package com.hst.sitescrapper.controller;

import com.hst.sitescrapper.model.api.ApiResponse;
import com.hst.sitescrapper.model.request.ScrapRequest;
import com.hst.sitescrapper.service.ScrapService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/scrap")
public class ScrapController {

    private Logger log = LoggerFactory.getLogger(getClass());

    @Autowired
    private ScrapService scrapService;

    @GetMapping
    public ApiResponse findWebSlices() {
        return ApiResponse.ok(scrapService.findScraps());
    }

    @PostMapping
    public ApiResponse createWebSlice(@RequestBody ScrapRequest scrapRequest) {
        log.info("New scrap request {}", scrapRequest);

        scrapService.createScrap(scrapRequest);

        return ApiResponse.ok();
    }



}

File: backend/src/main/kotlin/com/hst/scrapper/modules/scrap/ui/ScrapController.kt
package com.hst.scrapper.modules.scrap.ui

import com.hst.scrapper.global.model.api.ApiResponse
import com.hst.scrapper.modules.scrap.application.ScrapService
import com.hst.scrapper.modules.scrap.ui.request.CreateScrapRequest
import com.hst.scrapper.modules.scrap.ui.response.ScrapResponse
import org.springframework.web.bind.annotation.*

/**
 * @author dlgusrb0808@gmail.com
 */
@RestController
@RequestMapping("/api/scraps")
class ScrapController(
    val scrapService: ScrapService
) {

    @GetMapping
    fun getScraps(@RequestParam userId: Long): ApiResponse<List<ScrapResponse>> {
        return ApiResponse.success(scrapService.getScraps(userId))
    }

    @GetMapping("{id}")
    fun getScrap(@PathVariable id: Long): ApiResponse<ScrapResponse> {
        return ApiResponse.success(scrapService.getScrap(id))
    }

    @PostMapping
    fun createScrap(@RequestBody request: CreateScrapRequest): ApiResponse<Any?> {
        scrapService.createScrap(request)
        return ApiResponse.success()
    }

    @DeleteMapping("{id}")
    fun deleteScrap(@PathVariable id: Long): ApiResponse<Any?> {
        scrapService.deleteScrap(id)
        return ApiResponse.success()
    }

}